library/core/src/num/uint_macros.rs

```diff
@@ -3310,6 +3315,16 @@ macro_rules! uint_impl {
             let mut acc = 1;
 
             if intrinsics::is_val_statically_known(exp) {
+                // change of base:
+                // if base == 2 ** k, then
+                //    (2 ** k) ** n
+                // == 2 ** (k * n)
+                // == 1 << (k * n)
+                if self.is_power_of_two() {
```

Comment:
Isn't the is_val_statically_known() check on the wrong value here? You need a known base, not a known exponent?